src/app/auth/callback/page.tsx

```diff
@@ -0,0 +1,65 @@
+"use client"
+
+import { useBskyAuthContext } from "@/contexts"
+import { useEffect, useState } from "react"
+import { useRouter, useSearchParams } from "next/navigation"
+
+export default function OAuthCallback() {
+  const { bskyAuthClient, setAuthenticated, session, setSession, authenticated, setState } = useBskyAuthContext()
+  const [error, setError] = useState<string | null>(null)
+  const router = useRouter()
+  const searchParams = useSearchParams()
+
+  useEffect(() => {
+    const processCallback = async () => {
+      if (!bskyAuthClient) return;
+
+      try {
+        const hash = window.location.hash ? window.location.hash.substring(1) : ''
+
+        const params = hash
+          ? new URLSearchParams(hash)
+          : new URLSearchParams(searchParams.toString())
+
+        if (!params.has('code') || !params.has('state')) {
+          console.error("Missing required parameters in callback URL", params.toString())
+          throw new Error("Missing required OAuth parameters")
+        }
+
+        const result = await bskyAuthClient.callback(params)
+        if (!result) {
+          setError("Authentication failed. Please try again.")
+        }
+
+        console.log(`${result.session.sub} was successfully authenticated (state: ${result.state})`)
+        setAuthenticated(true)
```

Comment:
if we are going to do this manually we should either set the user profile from the session (if it's available) or use https://docs.bsky.app/docs/api/app-bsky-actor-get-profile to get it and set it